libminifi/src/SchedulingAgent.cpp

```diff
@@ -39,6 +40,36 @@ bool SchedulingAgent::hasWorkToDo(std::shared_ptr<core::Processor> processor) {
     return false;
 }
 
+void SchedulingAgent::enableControllerService(
+    std::shared_ptr<core::controller::ControllerServiceNode> &serviceNode) {
+
+  logger_->log_trace("Enabling CSN in SchedulingAgent %s",
+                     serviceNode->getName());
+  // reference the enable function from serviceNode
+  std::function<bool()> f_ex = [serviceNode] {
+    return serviceNode->enable();
+  };
+  // create a functor that will be submitted to the thread pool.
+  utils::Worker<bool> functor(f_ex);
+  // move the functor into the thread pool. While a future is returned
+  // we aren't terribly concerned with the result.
+  component_lifecycle_thread_pool_.execute(std::move(functor));
+}
+
+void SchedulingAgent::disableControllerService(
```

Comment:
I realize this occurs in Java land but was wondering about its applicability to MiNiFi in general.  Understand the need to handle the enabling process as there can be dependencies that require a certain sequence to occur, especially in the event that CS A depends on CS B.  Perhaps less for this PR but just a thought in general, should we tie the unscheduling/disabling of services into agent shutdown to allow appropriate cleanup without the process exiting abruptly?